<launch>
  <arg name="focal_len"  default="1.0"/>
  <arg name="field_size" default="0.25"/>
  <arg name="resolution" default="0.1"/>
  <arg name="cloud_name" default="/map_cloud"/>
  <arg name="pose_name" default="/camera_pose"/>

  <arg name="voc_file" default="$(find ORB_SLAM2)/Vocabulary/ORBvoc.bin"/>
  <arg name="settings_file" default="$(find ORB_SLAM2)/config/mono.yaml"/>
  <arg name="rgb_topic" default="/camera/rgb/image_raw"/>
  <arg name="depth_topic" default="/camera/depth_registered/image_raw"/>

  <node name="RGBDpub" pkg="ORB_SLAM2"  type="RGBDpub" 
        args = "$(arg voc_file) $(arg settings_file)" output="screen">

      <remap from="/camera/rgb/image_raw" to="$(arg rgb_topic)" />
      <remap from="/camera/depth_registered/image_raw" to="$(arg depth_topic)" />
       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />

       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <param name="map_file" type="string" value="map.bin" />
       <param name="settings_file" type="string" value="$(find ORB_SLAM2)/config/kinect.yaml" />
       <param name="voc_file" type="string" value="$(find ORB_SLAM2)/Vocabulary/ORBvoc.txt" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />

  </node>

  <node name="Explore_Node" pkg="ORB_SLAM2"  type="Explore_Node" 
        args = "$(arg focal_len) $(arg field_size) $(arg resolution) $(arg cloud_name) $(arg pose_name)  " output="screen"/>


</launch>